### PR TITLE
Use shared SearchInputView in ViewfinderView

### PR DESCRIPTION
**Pull request by ianguerin**
Support additional callbacks within SearchInputView for keydown and keyup events

shared-input-field

### CODE REVIEW

File: src/js/views/maps/viewfinder/ViewfinderView.js [hunk 1]
```diff
@@ -216,6 +184,34 @@ define(
...
 
+      /**
+       * Render the SearchInputView. 
+       */
+      renderSearchInput() {
+        this.searchInput = new SearchInputView({
+          placeholder: "Enter coordinates or areas of interest",
+          search: text => {
+            this.viewfinderModel.search(text);
+            // TODO(ianguerin): set error if results are empty.
```
Comment (yvonnesjy) [inline src/js/views/maps/viewfinder/ViewfinderView.js:195]: Is this for future or a reminder for this PR?

Comment (ianguerin) [reply src/js/views/maps/viewfinder/ViewfinderView.js]: No, just an out of date Todo, removing

File: src/css/map-view.css [hunk 2]
```diff
@@ -1323,7 +1323,7 @@ other class: .ui-slider-range */
     border: 1px solid var(--portal-col-bkg-active);
     border-radius: 4px;
     box-sizing: border-box;
-    color: var(--map-col-text);
+    color: var(--map-col-text-body);
```
Comment (yvonnesjy) [inline src/css/map-view.css:1326]: --map-col-text-body is only defined when a theme is used, while some portals/main repos don't use themes (you can test on ADC or DRP). do we need a fallback color for those use cases?

same question for other color variables.

**Review by robyngit — CHANGES_REQUESTED** [review]
Looks great!!! I added some suggestions where I thought we migth want to check that elements are defined before calling methods on them. As it is now, if one of the getSomeElement methods comes back undefined, then the rest of the Toolbar fails to render. Feel free to use a try/catch block or some other approach, just wanted to point that out.

**Comment by ianguerin** [comment]
> Looks great!!! I added some suggestions where I thought we migth want to check that elements are defined before calling methods on them. As it is now, if one of the getSomeElement methods comes back undefined, then the rest of the Toolbar fails to render. Feel free to use a try/catch block or some other approach, just wanted to point that out.

jQuery selectors don't return undefined/null when there's no element, so I don't think that the optional chaining will have any effect. I will, however, add `|| ''` to the getInputValue function so that it always returns a string

**Review by robyngit — APPROVED** [review]

**Pull request merged**